util/check_tool_requirements.py

```diff
@@ -47,17 +102,28 @@ def check_version(tool_name, required_version, actual_version):
 
 
 def main():
-    any_failed = False
+    # Get tool requirements
+    tool_requirements = read_tool_requirements()
 
-    if not check_version('verilator', __TOOL_REQUIREMENTS__['verilator'],
-                         get_verilator_version()):
-        any_failed = True
+    all_good = True
+    all_good &= check_version(tool_requirements,
+                              'verilator',
+                              get_verilator_version)
+    all_good &= check_version(tool_requirements,
+                              'edalize',
+                              lambda: pip3_get_version('edalize'))
 
-    if any_failed:
+    if not all_good:
         log.error("Tool requirements not fulfilled. "
                   "Please update the tools and retry.")
         return 1
+
     return 0
 
+
 if __name__ == "__main__":
-    sys.exit(main())
+    try:
+        sys.exit(main())
+    except (RuntimeError, ValueError) as err:
+        sys.stderr.write('Error: {}\n'.format(err))
+        sys.exit(1)
```

Comment:
But they aren't internal program errors: they are things that can happen if you edit a different file (`tool_requirements.py`). If you believe we shouldn't propagate the "aargh! stop now" logic using an exception, I can change the logic to do it C-style with return values.

Reply:
I see tool_requirements.py as an internal file to the tool. It's intentionally not a full config file, it's just a couple constants which happen to live in another file. Otherwise you end up with with more config file parsing/validation logic than you have "business logic."